src/TypesFinder/FindTypeFromAst.php

```diff
@@ -19,10 +19,16 @@ class FindTypeFromAst
      */
     public function __invoke($astType, LocatedSource $locatedSource, $namespace = '')
     {
-        $context = (new ContextFactory())->createForNamespace(
-            $namespace,
-            $locatedSource->getSource()
-        );
+        static $contexts = [];
```

Comment:
Would rather inject a context, then let the user decide whether to use a stateless or stateful service